src/uwtools/api/chgres_cube.py

```diff
@@ -4,21 +4,23 @@
 
 import datetime as dt
 from pathlib import Path
-from typing import Dict, Optional
+from typing import Dict, Optional, Union
 
 import iotaa as _iotaa
 
 import uwtools.drivers.support as _support
 from uwtools.drivers.chgres_cube import ChgresCube as _ChgresCube
+from uwtools.utils.api import ensure_data_source as _ensure_data_source
```

Comment:
The `ensure_data_source()` function 1. Enforces the requirement that a `None` data-source value in the API requires that `stdin_ok` is set, and 2. Converts `str` values to `Path` values to maintain the downstream invariant that `uwtools` uses `Path` objects internally.